solutions.Rmd

```diff
@@ -133,3 +154,142 @@ ggplot(data = mpg, aes(x=displ, y=hwy))+
+
+Using the mpg dataset create a bar chart showing a count of the dataset grouped by class of car.
+
+```{r, eval=FALSE}
+
+ggplot(data = mpg) + geom_bar(mapping = aes(x = class))
+
+```
+
+Now specify a y co-ordinate mapping to show the average fuel displacement of each class of car.
+
+  
+```{r, fig.height = 2.7}
+
+ggplot(data = mpg) + geom_bar(mapping = aes(x = class, y = displ), stat = "summary", fun.y = "mean")
+
+```
+
+---
+
+Add a fill to the bars to show the average split by the type of drive of each class of car. 
+
+  
+```{r, fig.height = 3}
+
+ggplot(data = mpg) + geom_bar(mapping = aes(x = class, y = displ, fill = drv), stat = "summary", 
+                              fun.y = "mean", position = "dodge")
+
+```
+
+---
+
+Extension: Manually change the colours of the fill being used. Also change the font size and angle of the x axis text labels so that these can be read in the viewer.
+
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) + 
+  geom_bar(mapping = aes(x = class, y = displ, fill = drv), stat = "summary", 
+           fun.y = "mean", position = "dodge") +
+  theme(axis.text.x = element_text(angle = 90, size = 8)) + 
+  scale_fill_manual(values = c("blue", "darkblue", "blueviolet"))
+
+```
+
+## 12
+
+Create a basic scatter plot showing highway fuel efficiency vs city fuel efficiency. Add a Wall Street Journal theme from the ggthemes package; and colour the points based on the number of cylinders the car has.
+
+Change the title and axis labels to something sensible based on what the chart is showing.
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) +
+  geom_point(mapping = aes(x = hwy, y = cty, colour = as.factor(cyl))) + 
+  theme_wsj() +
+  labs(title = "Fuel efficiency comparison between city and highway", 
+       x = "Highway efficiency", y = "City efficiency")
+
+```
+
+---
+
+Create a custom theme changing: major gridlines to blue; x axis font colour to red and y axis font colour to purple; move the legend position to the bottom of the plot.
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) +
+  geom_point(mapping = aes(x = hwy, y = cty, colour = as.factor(cyl))) + 
+  labs(title = "Fuel efficiency comparison between city and highway", 
+       x = "Highway efficiency", 
+       y = "City efficiency") +
+  theme(panel.grid.major = element_line(colour = "blue"),
+        axis.text.x = element_text(colour = "red"),
+        axis.text.y = element_text(colour = "purple"),
+        legend.position = "bottom")
+
+```
+
+---
+
+Try changing the x and y axis so that the scale starts at zero. 
+
+Change the title of the legend to 'Number of Cylinders' and re-name the labels with the text version of the number (e.g. 'four' instead of '4')
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) +
+  geom_point(mapping = aes(x = hwy, y = cty, colour = as.factor(cyl))) + 
+  labs(title = "Fuel efficiency comparison between city and highway", 
+       x = "Highway efficiency", 
+       y = "City efficiency") +
+  xlim(0,50) + ylim(0,40) +
```

Comment:
Good point. I've changed the solution so that the axes actually intercept and start at zero, and the plot background only covers the plotted area.
This makes the solution fairly complicated, but it is an extension exercise, and perhaps useful to show participants how to do this (as is possibly a fairly common task for plotting).